api/v1/pinning.go

```diff
@@ -234,6 +234,112 @@ func filterForStatusQuery(q *gorm.DB, statuses map[types.PinningStatus]bool) (*g
 	return q.Where("active or pinning or failed"), nil
 }
 
+// handleAddBatchedPins  godoc
+// @Summary      Add and pin objects in batches
+// @Description  This endpoint adds a pin to the IPFS daemon.
+// @Tags         pinning
+// @Accept		 json
+// @Produce      json
+// @Success      202	{object}  []types.IpfsPinStatusResponse
+// @Failure      500    {object}  util.HttpError
+// @in           202,default  string  Token "token"
+// @Param        pin          body      []types.IpfsPin  true   "Pin Body {[cid:cid, name:name]}"
+// @Param        ignore-dupes  query     string                   false  "Ignore Dupes"
+// @Param        overwrite	   query     string                   false  "Overwrite conflicting files in collections"
+// @Router       /pinning/batched-pins [post]
+func (s *apiV1) handleAddPins(c echo.Context, u *util.User) error {
+	ctx := c.Request().Context()
+
+	if err := util.ErrorIfContentAddingDisabled(s.isContentAddingDisabled(u)); err != nil {
+		return err
+	}
+
+	overwrite := false
+	if c.QueryParam("overwrite") == "true" {
+		overwrite = true
+	}
+
+	var pins []types.IpfsPin
+	if err := c.Bind(&pins); err != nil {
+		return err
+	}
+
+	var pinStatuses []*types.IpfsPinStatusResponse
+	for _, pin := range pins {
+		filename := pin.Name
```

Comment:
Instead of duplicating the code here can we delegate the call to handleAddPin?

At the very least extract out the duplicate logic in the same function.

Reply:
Agreed, but I think this new functionality should be on api v2 instead of v1. All new functionalities of API should go into v2